### ISSUE TITLE
Unify metric parameters

### ISSUE THREAD

**Issue by vnmabus** [GAA-UAM/scikit-fda#182]
Some classes have 'pairwise_distance(lp_distance)' as default parameter of `metric` and other ones an string with the value "lp" or "l2" (e.g. the neighbors module). We should unify a criterion at some point.

I think using strings as default arguments is clearer, and that's what is used in sklearn.

```
>>> FuzzyCMeans(..., metric=<function pairwise_distance.<locals>.pairwise at 0x11831a620>, ...)
>>> FuzzyCMeans(..., metric='l2', ...)
```

_Originally posted by @pablomm in https://github.com/GAA-UAM/scikit-fda/pull/179_

**Comment by vnmabus** [comment]
We are now using objects or functions for distances instead of strings whenever possible.
